music-journal/components/ListMemories.js

```diff
@@ -1,68 +1,33 @@
-
+import { useEffect, useState } from 'react';
 import { Row, Col, Table } from 'react-bootstrap/';
 
-
+//TODO: Move to services
 const fetcher = (url) =>
   fetch(url, {
     method: 'GET',
-  }).then((res) => res.json());
+}).then((res) => res.json());
 
+const ListMemories = ({ firebaseToken }) => {
+  const [memories, setMemories] = useState({}); 
 
-class ListMemories extends React.Component {
-  static async getInitialProps(ctx) {
-    const { firebaseToken } = cookies(ctx);
-    if (!firebaseToken) {
-      return {}
-    }
-    return { firebaseToken };
-  }
-
-
-  constructor(props) {
-    super(props);
-    this.state = {};
-  }
-
-  populateMemoriesTable(memories) {
-    document.getElementById("nomemories").remove();
-    let tableBody = document.getElementById("memoriestablebody");
-    let tableLength = 5;
-    if(memories.length < 5) {
-      tableLength = memories.length;
-    }
-    for (let i = 0; i < tableLength; i++) {
-      let tableRow = document.createElement('tr');
-      let songColumn = document.createElement('td');
-      let memoryColumn = document.createElement('td');
-      songColumn.innerHTML = memories[i].song;
-      memoryColumn.innerHTML = memories[i].text;
-      tableRow.appendChild(songColumn);
-      tableRow.appendChild(memoryColumn);
-      tableBody.appendChild(tableRow);
-    }
-  }
-
-  componentDidMount() {
+  useEffect(() => {
```

Comment:
hope this works, think I _may_ have experimented with useEffect and couldn't get it to work

Reply:
the main thing with useEffect is the second argument which lets it know when to run the function again. If you have an empty array it just does it when the component loads. Looks like it works.

Reply:
TIL!